test/module/irohad/ametsuchi/postgres_executor_test.cpp

```diff
@@ -1996,5 +1996,204 @@ namespace iroha {
+                  account_id);
+
+      std::vector<std::string> query_args{"doge@noaccount", "key", "value"};
+      CHECK_ERROR_CODE_AND_MESSAGE(cmd_result, 3, query_args);
+    }
+
+    /**
+     * @given command
+     * @when trying to set kv and then set kv1 with correct old value
+     * @then kv1 is set
+     */
+    TEST_F(CompareAndSetAccountDetail, ValidOldValue) {
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+              account_id, "key", "value", boost::none)));
+
+      auto kv = sql_query->getAccountDetail(account_id);
+      ASSERT_TRUE(kv);
+      ASSERT_EQ(kv.get(), "{\"id@domain\": {\"key\": \"value\"}}");
+
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+              account_id,
+              "key",
+              "value1",
+              boost::optional<
+                  shared_model::interface::types::AccountDetailValueType>(
+                  "value"))));
+      auto kv1 = sql_query->getAccountDetail(account_id);
+      ASSERT_TRUE(kv1);
+      ASSERT_EQ(kv1.get(), "{\"id@domain\": {\"key\": \"value1\"}}");
+    }
+
+    /**
+    * @given command
+    * @when trying to set kv and then set kv1 with incorrect old value
+    * @then corresponding error code is returned
+    */
+    TEST_F(CompareAndSetAccountDetail, InvalidOldValue) {
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+              account_id, "key", "value", boost::none)));
+
+      auto kv = sql_query->getAccountDetail(account_id);
+      ASSERT_TRUE(kv);
+      ASSERT_EQ(kv.get(), "{\"id@domain\": {\"key\": \"value\"}}");
+
+      auto cmd_result =
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+              account_id, "key", "value1", boost::optional<shared_model::interface::types::AccountDetailValueType>("oldValue")));
+
+      std::vector<std::string> query_args{account_id, "key", "value1", "oldValue"};
+      CHECK_ERROR_CODE_AND_MESSAGE(cmd_result, 4, query_args);
+    }
+
+    /**
+     * @given Two commands
+     * @when trying to set kv and kv1 simultaneously
+     * @then kv is set and corresponding error code is returned for kv1
+     */
+    TEST_F(CompareAndSetAccountDetail, TwoSimultaneousCommands) {
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+              account_id, "key", "value", boost::none)));
+
+      auto cmd_result =
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+              account_id, "key", "value1", boost::none));
+
+      std::vector<std::string> query_args{account_id, "key", "value1"};
+      CHECK_ERROR_CODE_AND_MESSAGE(cmd_result, 4, query_args);
+
+      auto kv = sql_query->getAccountDetail(account_id);
+      ASSERT_TRUE(kv);
+      ASSERT_EQ(kv.get(), "{\"id@domain\": {\"key\": \"value\"}}");
```

Comment:
[raw strings](https://en.cppreference.com/w/cpp/language/string_literal) would be easier to read and modify

Reply:
Fixed